### DISCUSSION TITLE
SSH Certificates for signing commits

### DISCUSSION THREAD

**Discussion by lpcalisi** [community/community#55204] [Enterprise]
### Select Topic Area

Question

### Select Feature Area

- [ ] Issues
- [X] Projects

### Body

Does using SSH certificates for signing commits make sense?

By now, we have the possibility to authenticate to Github Enterprise (from git) using SSH certificates, but .. if you want to sign commits with strong authentication, you will need some tool that provides, for example short term x509 certificates (like sigstore)

I think supports SSH certificates as an authentication and sign method would be an opportunity to provide authentication and integrity in a single way.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by lpcalisi** [comment]
git 2.34 supports sign & verificate commits with SSH certificates.

`gpg.ssh.allowedSignersFile` git option is based on openssh [ALLOWED SIGNERS](https://man.openbsd.org/ssh-keygen.1#ALLOWED_SIGNERS) to verificate commit  signed with SSH certificates

```shell
$ git show --show-signature
commit 2ab6a304e0dc763f06bf6e3e469dd57bca95d812 (HEAD -> ssh-ca-sign)
Good "git" signature for lcalisi with RSA-CERT key SHA256:mm8XrAmALO+oSCmTM/aXaKVH4tzyuFa70vubXeQn5PM
Author: Lucas
Date:   Thu May 11 23:19:57 2023 -0300

    sign with ephemeral SSH certificate
```

**Reply by lpcalisi** [reply]
@davidstaheli

**Comment by brian6932** [comment]
This is a thing, you just have to add the ssh key again as a `signing key`

**Reply by Jackbennett** [reply]
It doesn't accept the `-cert.pub` format and the `ssh.pub` key is already in use. My certificate signed commits are "unverified" even with the enterprise CA added.
